p2panda-rs/src/entry/tests.rs

```diff
@@ -65,6 +70,104 @@ fn sign_and_encode_roundtrip(#[case] entry: Entry, key_pair: KeyPair) {
     assert!(sign_and_encode(&entry_second, &key_pair).is_ok());
 }
 
+#[rstest]
+#[should_panic(expected = "InvalidHexCharacter { c: 'Z', index: 9 }")]
+#[case::invalid_hex_string("123456789Z")]
+#[should_panic(expected = "OddLength")]
+#[case::another_invalid_hex_string(":{][[5£$%*(&*££  ++`/.")]
+#[case::should_not_have_skiplink(entry_signed_encoded_unvalidated(
+    entry_unvalidated(
+        1,
+        1,
+        None,
+        Some(random_hash()),
+        Some(operation(Some(operation_fields(default_fields())), None, None))
+    ),
+    random_key_pair()
+))]
+#[should_panic(expected = "DecodePayloadHashError { source: DecodeError }")]
+#[case::should_not_have_backlink(entry_signed_encoded_unvalidated(
+    entry_unvalidated(
+        1,
+        1,
+        Some(random_hash()),
+        None,
+        Some(operation(Some(operation_fields(default_fields())), None, None))
+    ),
+    random_key_pair()
+))]
+#[should_panic(expected = "DecodePayloadHashError { source: DecodeError }")]
+#[case::should_not_have_backlink_or_skiplink(
+    entry_signed_encoded_unvalidated(
+        entry_unvalidated(
+            1,
+            1,
+            Some(DEFAULT_HASH.parse().unwrap()),
+            Some(DEFAULT_HASH.parse().unwrap()),
+            Some(operation(Some(operation_fields(default_fields())), None, None))
+        ),
+        random_key_pair()
+    ),
+)]
+#[should_panic(expected = "DecodeBacklinkError { source: DecodeError }")]
+#[case::missing_backlink(entry_signed_encoded_unvalidated(
+    entry_unvalidated(
+        2,
+        1,
+        None,
+        None,
+        Some(operation(Some(operation_fields(default_fields())), None, None))
+    ),
+    random_key_pair()
+))]
+#[should_panic(expected = "DecodeBacklinkError { source: DecodeError }")]
+#[case::missing_skiplink(entry_signed_encoded_unvalidated(
+    entry_unvalidated(
+        8,
+        1,
+        Some(random_hash()),
+        None,
+        Some(operation(Some(operation_fields(default_fields())), None, None))
+    ),
+    random_key_pair()
+))]
+#[should_panic(expected = "DecodePayloadHashError { source: DecodeError }")]
+#[case::should_ommit_skiplink_when_same_as_backlink(
+    entry_signed_encoded_unvalidated(
+        entry_unvalidated(
+            14,
+            1,
+            Some(DEFAULT_HASH.parse().unwrap()),
+            Some(DEFAULT_HASH.parse().unwrap()),
+            Some(operation(Some(operation_fields(default_fields())), None, None))
+        ),
+        random_key_pair()
+    )
+)]
+#[should_panic(expected = "SOME PANIC MESSAGE")] // TODO: do we expect this case to fail?
+#[case::skiplink_and_backlink_should_not_be_the_same(entry_signed_encoded_unvalidated(
+    entry_unvalidated(
+        13,
+        1,
+        Some(DEFAULT_HASH.parse().unwrap()),
+        Some(DEFAULT_HASH.parse().unwrap()),
+        Some(operation(Some(operation_fields(default_fields())), None, None))
+    ),
+    random_key_pair()
+))]
```

Comment:
Would we expect this case to panic? 

It's an entry which MUST have a backlink and skiplink, and they should be different (but they're the same).